cognite/client/data_classes/simulators/simulators.py

```diff
@@ -0,0 +1,322 @@
+            if "modelTypes" in resource
+            else None,
+            step_fields=SimulatorStep._load_list(resource["stepFields"], cognite_client)
+            if "stepFields" in resource
+            else None,
+            unit_quantities=SimulatorQuantity._load_list(resource["unitQuantities"], cognite_client)
+            if "unitQuantities" in resource
+            else None,
+        )
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        output = super().dump(camel_case=camel_case)
+        if isinstance(self.model_types, list) and all(isinstance(item, SimulatorModelType) for item in self.model_types):
+            output["modelTypes" if camel_case else "model_types"] = [item.dump(camel_case=camel_case) for item in self.model_types]
+        if isinstance(self.step_fields, list) and all(isinstance(item, SimulatorStep) for item in self.step_fields):
+            output["stepFields" if camel_case else "step_fields"] = [item.dump(camel_case=camel_case) for item in self.step_fields]
+        if isinstance(self.unit_quantities, list) and all(isinstance(item, SimulatorQuantity) for item in self.unit_quantities):
+            output["unitQuantities" if camel_case else "unit_quantities"] = [item.dump(camel_case=camel_case) for item in self.unit_quantities]
+
+        return output
+
+
+@dataclass
+class SimulatorRoutineStep(CogniteObject):
+    step_type: str
+    arguments: dict[str, Any]
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            step_type=resource["stepType"],
+            arguments=resource["arguments"],
+        )
+
+
+@dataclass
+class SimulatorUnitEntry(CogniteObject):
+    label: str
+    name: str
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            label=resource["label"],
+            name=resource["name"],
+        )
+
+
+@dataclass
+class SimulatorStepOption(CogniteObject):
+    label: str
+    value: str
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            label=resource["label"],
+            value=resource["value"],
+        )
+
+
+@dataclass
+class SimulatorModelType(CogniteObject):
+    name: str
+    key: str
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            name=resource["name"],
+            key=resource["key"],
+        )
+
+    @classmethod
+    def _load_list(
+        cls, resource: dict[str, Any] | list[dict[str, Any]], cognite_client: CogniteClient | None = None
+    ) -> SimulatorModelType | list[SimulatorModelType]:
+        if isinstance(resource, list):
+            return [cls._load(res, cognite_client) for res in resource]
+
+        return cls._load(resource, cognite_client)
```

Comment:
```suggestion
    @classmethod
    def _load_list(
        cls, resource: list[dict[str, Any]], cognite_client: CogniteClient | None = None
    ) -> list[SimulatorModelType]:
        return [cls._load(res, cognite_client) for res in resource]
```
this one and others should be fixed to return the array only.
but in this case I don't think we need the method and can just do it inline

Reply:
Btw, the pre-commit hook's type checking did not allow me to simply return an array, so i added a few extra lines to this part